src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -21,47 +71,608 @@ const ActorInit En_Box_InitVars = {
+            sp4c_3f.z += f0 * 20.0f * Math_Coss(this->dyna.actor.posRot.rot.y + 0x4000);
+            f0 = Math_Rand_ZeroOne();
+            f0 = f0 + f0 - 1.0f;
+            sp40_3f.x = f0 * 1.6f * Math_Sins(this->dyna.actor.posRot.rot.y);
+            sp40_3f.y = 1.8f;
+            sp40_3f.z = f0 * 1.6f * Math_Coss(this->dyna.actor.posRot.rot.y);
+        }
+        func_8002AA44(globalCtx, &sp4c_3f, &sp40_3f, &sp34_3f, 0x96);
+    }
+}
+
+void EnBox_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBox* this = THIS;
+
+    if (this->movementFlags & ENBOX_MOVE_STICK_TO_GROUND) {
+        this->movementFlags &= ~ENBOX_MOVE_STICK_TO_GROUND;
+        EnBox_ClipToGround(this, globalCtx);
+    }
+
+    (this->actionFunc)(this, globalCtx);
+
+    if (!(this->movementFlags & ENBOX_MOVE_IMMOBILE)) {
+        Actor_MoveForward(&this->dyna.actor);
+        func_8002E4B4(globalCtx, &this->dyna.actor, 0.0f, 0.0f, 0.0f, 0x1C);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/EnBox_Init.s")
+    switch (this->type) {
+        case ENBOX_TYPE_SMALL:
+        case ENBOX_TYPE_6:
+        case ENBOX_TYPE_ROOM_CLEAR_SMALL:
+        case ENBOX_TYPE_SWITCH_FLAG_FALL_SMALL:
+            Actor_SetHeight(&this->dyna.actor, 20.0f);
+            break;
+        default:
+            Actor_SetHeight(&this->dyna.actor, 40.0f);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/EnBox_Destroy.s")
+    if ((this->dyna.actor.params >> 5 & 0x7F) == 0x7C && this->actionFunc == EnBox_Open &&
+        this->skelanime.animCurrentFrame > 45 && this->iceEffTimer < 100) {
+        func_809C9EF8(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C92F4.s")
+void EnBox_PostLimbDraw2(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
+    EnBox* this = THIS;
+    s32 unused;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C93F8.s")
+    if (limbIndex == 1) {
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_box.c", 1492),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        if (this->type != ENBOX_TYPE_DECORATED_BIG) {
+            gSPDisplayList((*gfx)++, D_060006F0);
+        } else {
+            gSPDisplayList((*gfx)++, D_06000AE8);
+        }
+    } else if (limbIndex == 3) {
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_box.c", 1502),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        if (this->type != ENBOX_TYPE_DECORATED_BIG) {
+            gSPDisplayList((*gfx)++, D_060010C0);
+        } else {
+            gSPDisplayList((*gfx)++, D_06001678);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C94AC.s")
+Gfx* EnBox_EmptyDList(GraphicsContext* gfxCtx) {
+    Gfx* displayList;
+    Gfx* displayListHead;
```

Comment:
we havent done this project wide yet, but we plan to go with dList as a name consistently at some point, if youd like to change it now